counter/src/main/java/org/infinispan/counter/impl/manager/CounterConfigurationManager.java

```diff
@@ -84,7 +85,7 @@ public void start() {
       Map<String, CounterConfiguration> persisted = storage.loadAll();
       persisted.forEach((name, cfg) -> stateCache.putIfAbsent(stateKey(name), cfg));
       counterCacheStarted.set(false);
-      stateCache.addListener(listener, new ScopeFilter(COUNTER_SCOPE));
+      stateCache.addFilteredListener(listener, new ScopeFilter(COUNTER_SCOPE), null, Util.asSet(CacheEntryCreated.class, CacheEntryModified.class));
```

Comment:
Shouldn't need to specify the event types, as this is handled by the annotation itself. This is for remote listeners which we don't have access to the annotations at this point.

Reply:
I didn't realise it was for embedded. I figured that it might be more efficient explicitly stating them ... I didn't look into the internals. I'll update now :+1: